esphome/components/adc/adc_sensor.h

```diff
@@ -28,6 +28,21 @@ static const adc_atten_t ADC_ATTEN_DB_12_COMPAT = ADC_ATTEN_DB_11;
 #endif
 #endif  // USE_ESP32
 
+enum class SamplingMode : uint8_t { AVG = 0, MIN = 1, MAX = 2 };
+const char *sampling_mode_str(SamplingMode mode);
```

Comment:
```suggestion
const char *const sampling_mode_str(SamplingMode mode);
```
Maybe just make the pointer itself const also.